parser.y

```diff
@@ -10,13 +11,21 @@
 #include "operator.hpp"
 #include "type.hpp"
 
+namespace {
+/// @brief Resolved the innermost unknown type to the given type.
+/// @note If the type is a primitive type, the resolved type is returned.
```

Comment:
This comment is different from the implementation, where `resolved_type` is returned, only if `type` is a primitive `unknown type`. If `type` is a `PrimType::int`, do you intend to return `resolved_type`?

Reply:
If type is a primitive type to int, the resolve function should not even be called, which is why I include an assertion in the beginning of the implementation.